### PR TITLE
Use Kotlin Poet for generating the hints. 

### PR DESCRIPTION
**Pull request by vRallev**
This is a clean up PR. I believe this way the generated code is easier to read, because proper imports are used.

### CODE REVIEW

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt [hunk 1]
```diff
@@ -294,23 +293,21 @@ internal class BindingModuleGenerator(
...
-          .toList()
+        it.findChildrenByClass(KtProperty::class.java).toList()
       }
       .mapNotNull { ktProperty ->
-        (
-          (ktProperty.initializer as? KtClassLiteralExpression)
-            ?.firstChild as? KtDotQualifiedExpression
-          )
-          ?.text
-          ?.let { FqName(it) }
```
Comment (vRallev) [inline compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt:313]: This wasn't great, it was pretty much hardcoded that a fully qualified name is used for `val prop = abc.def.Class::class`.

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/ContributesBindingGenerator.kt [hunk 2]
```diff
@@ -42,35 +48,44 @@ internal class ContributesBindingGenerator : CodeGenerator {
...
-            '_'
-          )}$SCOPE_SUFFIX: kotlin.reflect.KClass<$scope> = $scope::class
-          """.trimIndent()
-        file.writeText(content)
+          FileSpec.buildFile(generatedPackage, propertyName) {
+            addProperty(
+              PropertySpec
+                .builder(
+                  name = propertyName + REFERENCE_SUFFIX,
+                  type = KClass::class.asClassName().parameterizedBy(className)
```
Comment (christiandeange) [inline compiler/src/main/java/com/squareup/anvil/compiler/codegen/ContributesBindingGenerator.kt:64]: nice

**Review by christiandeange — APPROVED** [review]

**Review by FrostRocket — APPROVED** [review]
Nice cleanup!

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/ContributesBindingGenerator.kt (hunk 2 above)
Comment (FrostRocket) [reply compiler/src/main/java/com/squareup/anvil/compiler/codegen/ContributesBindingGenerator.kt:64]: Chain is a lot easier to follow 👍

**Pull request merged**